From b86dd6fc7b327db58fa6de4190e6269e41101fcc Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jiri@resnulli.us>
Date: Tue, 1 Jan 2013 03:30:14 +0000
Subject: [PATCH 465/858] net: call add_device_randomness() only after
 successful mac change

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit f652151640344f8faee21fcf702b4fd46b8fe70b)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 net/core/dev.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/net/core/dev.c b/net/core/dev.c
index 98a5a88..acf0558 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -5072,10 +5072,11 @@ int dev_set_mac_address(struct net_device *dev, struct sockaddr *sa)
  if (!netif_device_present(dev))
   return -ENODEV;
  err = ops->ndo_set_mac_address(dev, sa);
- if (!err)
-  call_netdevice_notifiers(NETDEV_CHANGEADDR, dev);
+ if (err)
+  return err;
+ call_netdevice_notifiers(NETDEV_CHANGEADDR, dev);
  add_device_randomness(dev->dev_addr, dev->addr_len);
- return err;
+ return 0;
 }
 EXPORT_SYMBOL(dev_set_mac_address);
 
-- 
1.8.3.1